### PR TITLE
Implementing TOML file for packaging

### LINKED ISSUE #36: Duplicated README files
There are two README files in the repository with different information. It would be better to consolidate the information in a single file. Can we move the text from [README.rst](https://github.com/openearth/aeolis-python/blob/AEOLIS_V2/README.rst) to README.md?

### LINKED ISSUE #107: In the near future we might want to use a TOML file to describe dependencies and build environments
The use of `septup.py` to define how a package should be built it is officially deprecated but still in use. So this is not urgent, but highly recommended.

https://guide.esciencecenter.nl/#/best_practices/language_guides/python?id=making-an-installable-package

### PR DESCRIPTION
**Pull request by manuGil**
- Implement pyproject.toml
- Change build-system from setuptools to hatchling
- Remove docs/ and test/ directories from source disbtirution
- Update Readme.md Release.md and Contiguting.md
- Modified model.py for fetching version from package metadata

Closes #107 
Fixes #36

### CODE REVIEW

**Review by niketagrawal — CHANGES_REQUESTED** [review]
Thanks for adding the changes. 

The publish-workflow.yml needs to be modified because the names of the source and wheel distributions created with the pyproject and hatchling based build system are different than the ones created by the existing setup.py based build system.

The names of the source and wheel distributions created by the `python -m build` command with these changes 
are `aeolis-2.1.1-py3-none-any.whl` and `aeolis-2.1.1.tar.gz`, however, [the publish-workflow.yml in main searches for the pattern AeoLiS-*.whl during installation test](https://github.com/openearth/aeolis-python/blob/a520985fb66e10a58ad3e1c5eec51722c05ae79c/.github/workflows/publish-release.yml#L39)

Can you please update the publish-workflow.yml as part of these changes?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -74,7 +74,7 @@ If you are a part of the AeoLiS team and have write access to the aeolis-python
 3. Set up a development environment on your computer by installing the AeoLiS in development mode with the following command: (Consider using a virtual environment for this purpose.)
 
     ```bash
-    pip install -e .
+    pip install -e .[dev]
```
[the remark this answers is not in the document]
Comment (manuGil) [reply CONTRIBUTING.md:77]: > Remove `-e` in the absence of setup.py.
> 
> `pip install .[dev]` is the correct command as mentioned in your changes to the README.md file.

`pip install -e .[dev] works for me, while in the root directory of the repo (where the `pyproject.toml`) is located. Can you try one more time, please?

[the remark this answers is not in the document]
Comment (niketagrawal) [reply CONTRIBUTING.md:77]: Thank you for the hint to update pip.
The command `pip install -e .[dev]` works after updating pip to the latest version (23.1.2).

**Review by niketagrawal — APPROVED** [review]

**Comment by manuGil** [comment]
@niketagrawal I completed the changes. I will merge this PR

**Pull request merged**